### DISCUSSION TITLE
Reset to Bootloader

### DISCUSSION THREAD

**Discussion by hiszd** [Rahix/avr-hal#417] [Q&A]
I am using a atmega32u4 with the caterina bootloader on it.
I am not sure how to go about this using this hal.
In C I have seen it done where they write 0x7777 to the register at 0x0800, then enable the watchdog timer with a short delay.
Can someone help with this?

**Comment by Rahix** [comment]
I am missing context on your question but if you just want to write a value to an address, try this:

```rust
unsafe { core::ptr::write_volatile(0x0800 as *mut u16, 0x7777) };
```
